src/services/DataTools.js

```diff
@@ -0,0 +1,70 @@
+const ColorConsole = require('../services/ColorConsole');
+// const ConfigService = require('./services/Config/ConfigService');
+// TODO use ConfigService getTestRailStatusSkipped, getTestRailStatusFailed, 
+//    and getTestRailStatusPassed Instead of it 2, 5, 1
+
+const RESULT_STATUS_FAILED = 5;
+const RESULT_STATUS_SKIP = 2;
+// const RESULT_STATUS_PASS = 1;
+
+
+class DataTools {
+
+  
+  sayHiCyTr() {
+    ColorConsole.debug('Hi from DataTools.sayHiCyTr' );
+  }
+
+  /**
+   *
+   * @param {cytrPostData}
+   * @returns {*}
+   */
+  aggregateDuplicateResults(cytrPostData) {
+    // Replace multiple occurrences of test cases with one output summarizing case, aggregation rules are;
+    // Fail trumps all, anything with a failed leads to the first failed item is the one emitted
+    // If no mixed outcome, the same result is passed through
+    // Pass trumps skip
+    // If all test pass. The last comment is the one emitted.
+    // if all pass or skip modify the comment statement with the prefix of `Summarization of {n} Cypress tests` 
+    const emitPostData = []
+    const mergedResults = {} // this is a map of case_id to merged result object
+
+    cytrPostData.forEach((postData) => {
+      const { case_id, status_id, comment, elapsed, screenshotPaths } = postData;
+      if (Object.hasOwnProperty.call(mergedResults, case_id)) {
+        if (mergedResults[case_id].status_id !== RESULT_STATUS_FAILED) { // If one case failed don't process additional cases, Fail trumps all
+          mergedResults[case_id].count += 1;
```

Comment:
Do you actually want to increment regardless of the status? 

If the order of the results was "pass, pass, fail" the count would be 3 but if it was "fail, pass, pass" it would be 1 I think.  

If you move this line above line 36 the count will always increment regardless of the order of the results.

Reply:
I'm not really sure what the right thing to do is here. We might count only the number of passes currently we count the number of passes or skips. We don't count how many elements there are after we hi  fail. 

I think this is an area where we can improve the question is, how do we count should be only count consecutive passes that seems the clearest the code doesn't do that as currently written.

Reply:
Let's chat on Wednesday and we can come up with a strategy.

Reply:
After thinking about the count business, the complete way of handling it would be counting individually how many pass fail and skips occur on a per TR case basis, the current code doesn't support this I'd have to rewrite it, but it shouldn't be too difficult. Alternatively, I would drop the business of augmenting the comment with notation at all.
Possible new summary text:
`Summarization {case_id} has {m) passes {n} failures and {i} skips`